exporter/coralogixexporter/traces_client.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/pdata/ptrace/ptraceotlp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type tracesExporter struct {
+	// Input configuration.
+	config *Config
+
+	traceExporter ptraceotlp.Client
+	clientConn    *grpc.ClientConn
+	metadata      metadata.MD
+	callOptions   []grpc.CallOption
+
+	settings component.TelemetrySettings
+
+	// Default user-agent header.
+	userAgent string
+}
+
+func newTracesExporter(cfg config.Exporter, set component.ExporterCreateSettings) (*tracesExporter, error) {
+	oCfg := cfg.(*Config)
```

Comment:
```suggestion
	oCfg, ok := cfg.(*Config)
	if !ok {
		return nil, fmt.Errorf("invalid config exporter, expect type: %T, got: %T". &Config{}, cfg)
	}
```
mh.. We could check the type here to avoid some trouble in the future. What do you think?